sql-plugin/src/main/scala/com/nvidia/spark/SQLPlugin.scala

```diff
@@ -26,6 +26,9 @@ import org.apache.spark.internal.Logging
  * To enable this plugin, set the config "spark.plugins" to `com.nvidia.spark.SQLPlugin`
  */
 class SQLPlugin extends SparkPlugin with Logging {
-  override def driverPlugin(): DriverPlugin = new RapidsDriverPlugin
-  override def executorPlugin(): ExecutorPlugin = new RapidsExecutorPlugin
+  override def driverPlugin(): DriverPlugin =
+    ShimLoader.newInstanceOf("com.nvidia.spark.rapids.RapidsDriverPlugin")
```

Comment:
I am not thrilled with a reflection like API when the number of classes is small and we could have a dedicated API for each. Is there a reason this needs to be reflection based?

Reply:
The purpose is to prevent the caller's classloader being used at the call site. The first entry point in the code we can control must be reflection-based unless and until we can push the classloader changes upstream to Apache Spark. Otherwise, we will end up with even more reflection calls.

Reply:
Then can we try to limit all of the reflection to be self contained in ShimLoader?  having newInstanceOf be private forces people to put the reflection code in the ShimLoader and helps us keep it all in once place.